WIP: Move `recordSignupComplete` to response handler of `/sites/new`

gmovr opened the pull request:
#### Changes proposed in this Pull Request

* Ref pau2Xa-3E9-p2 the event `calypso_new_user_site_creation` fires in some flows that do not create a new site. 
* This PR attempts to fix this, by moving the call to `recordSignupComplete` inside the response handler of `/sites/new/` (where we know a new site has been created). This is similar to how `recordRegistration` is called when a new user is created.
* I've also added a check to see if current sites <= 1, to prevent this event from firing on any subsequent site creations from the same user.

#### Testing instructions
TBD, but basic manual testing flow:
* Create a new user via /start/ 
* Create a new site by following the signup flow
* **Expected**: `calypso_new_user_site_creation` should fire
* Create a new site under the same account
* **Expected**: `calypso_new_user_site_creation` should not fire
* Create a new user via https://wordpress.com/start/domain
* **Expected**: `calypso_new_user_site_creation` should not fire

niranjan-uma-shankar commented on client/lib/signup/step-actions/index.js:
```diff
@@ -318,6 +343,17 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 				isFreeThemePreselected,
 				themeSlugWithRepo
 			);
+			recordSignupComplete( {
```
In the [function definition](https://github.com/Automattic/wp-calypso/blob/6b3c29ae140352b47ba3f8bc2c5ed41b5b5b34d6/client/lib/analytics/signup.js#L27) of `recordSignupComplete()`, I feel we should decouple the Tracks calls to `calypso_signup_complete` and `calypso_new_user_site_creation`. In other words, we should have a separate function to record `calypso_new_user_site_creation`.

The reason for this is that `calypso_signup_complete` records the completion of a signup flow. We have many signup flows that don't lead to site creation, for example the `https://wordpress.com/start/account` flow or `/start/domain` flow. Per this PR's change to record these events as a response handler to `/sites/new`, then event `calypso_signup_complete` event will not get recorded in such flows. 

I think we should continue to keep the changes in `client/signup/main.jsx` file, but remove `calypso_new_user_site_creation` from `recordSignupComplete()`, and then record site creation alone in the "/sites/new" callback. Does this make sense to you?

niranjan-uma-shankar commented on client/lib/signup/step-actions/index.js:
```diff
@@ -257,8 +262,29 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
+		( dependencies.cartItem || dependencies.domainItem || dependencies.themeItem )
+	);
+
+	const isNew7DUserSite = !! (
```
A minor comment, but I suggest removing the newline after each variable definition.

niranjan-uma-shankar reviewed (commented):
Thanks for working on this! It's shaping up well.

gmovr committed:
Added new function splitting out the tracking of calypso_new_user_site_creation from calypso_signup_complete.

gmovr committed:
Removed tracking call from function, added a conditional to check site count before firing calypso_new_user_site_creation

gmovr committed:
Some cleanup + using existing selector

niranjan-uma-shankar commented on client/lib/signup/step-actions/index.js:
```diff
@@ -281,8 +285,8 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 	}
-
 	const locale = getLocaleSlug();
+	const isNewUser = getCurrentUserSiteCount( state ) <= 1;
```
Should we also check if the user is < 7 days old?

gmovr replied:
You're right! I'll update the PR with a <7 day check 👍

gmovr replied:
Updated, but I'm not 100% sure about the conditional - since we're here firing it inside the response handler, so I think we should check the site count, and then if they registered within the last 7 days. 

Because in the current code, in `main.jsx` we also check for site slug and a few more things, but we shouldn't need that here, since we can be pretty confident a site is being created.

I'll do some more testing to.

niranjan-uma-shankar replied:
> Because in the current code, in main.jsx we also check for site slug and a few more things, but we shouldn't need that here, since we can be pretty confident a site is being created.

Right, since the site is already created, we don't need to check for siteSlug, so checking for site count and if it's a new 7D user should be good.

gmovr committed:
Additional check: user registered within days (0,7)
Makes the conditional for firing `calypso_new_user_site_creation` the same as current prod code.

niranjan-uma-shankar commented on client/lib/signup/step-actions/index.js:
```diff
@@ -288,8 +293,10 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
-
 	const locale = getLocaleSlug();
+	const isNew7dUser = !! (
+		getCurrentUserSiteCount( state ) <= 1 && isUserRegistrationDaysWithinRange( state, null, 0, 7 )
```
When I tested, `isNew7dUser` is getting a falsey value, so `recordNewUserSiteCreated()` is not getting fired.
